server/src/test/java/org/elasticsearch/index/query/IntervalQueryBuilderTests.java

```diff
@@ -49,7 +49,18 @@
 
     @Override
     protected IntervalQueryBuilder doCreateTestQueryBuilder() {
-        return new IntervalQueryBuilder(STRING_FIELD_NAME, createRandomSource());
+        while (true) {
```

Comment:
Can we instead limit the number the number of recursion that we allow in `createRandomSource()` ? More than 2-3 levels seem overkill to test.

Reply:
I pushed 21d9f121be19a48d525bb7e7f576e068409f7766